From 37b31373b1408c299270e3134b88eff4528ca75a Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 16 Jan 2014 17:31:57 +0100
Subject: [PATCH 10/10] misc: Use wipefs -a in misc.wipefs() helper

Currently we only wipe the first signature we can find with wipefs.
However wipefs has an ability to wipe all the signatures of given type
from the device (yes this is once again about btrfs) with option '-a'.
So use that instead. This also simplify the wipefs helper a lot.

Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---
 ssmlib/misc.py                | 14 ++------------
 tests/unittests/test_btrfs.py | 34 +++++++++++++++++-----------------
 2 files changed, 19 insertions(+), 29 deletions(-)

diff --git a/ssmlib/misc.py b/ssmlib/misc.py
index 3d99c07..915cad1 100644
--- a/ssmlib/misc.py
+++ b/ssmlib/misc.py
@@ -363,18 +363,8 @@ def get_dmnumber(name):
 def wipefs(device, signatures):
     if type(signatures) is not list:
         signatures = [signatures]
-    command = ['wipefs', '-p', device]
-    output = run(command)[1]
-    offset = []
-    for line in output[1:].split('\n'):
-        if not line:
-            continue
-        array = line.split(",")
-        if array[-1] in signatures:
-            offset.extend(['--offset', array[0]])
-    if len(offset) > 1:
-        command = ['wipefs'] + offset + [device]
-        run(command)
+    command = ['wipefs', '-a', '-t', ','.join(signatures), device]
+    run(command)
 
 
 def humanize_size(arg):
diff --git a/tests/unittests/test_btrfs.py b/tests/unittests/test_btrfs.py
index 2fb04bc..2b6cd8e 100644
--- a/tests/unittests/test_btrfs.py
+++ b/tests/unittests/test_btrfs.py
@@ -167,12 +167,12 @@ class BtrfsFunctionCheck(MockSystemDataSource):
         self._addPool('my_pool', ['/dev/sdc2', '/dev/sdc3', '/dev/sdc1'])
 
         # remove volume
-        self._checkCmd("ssm remove default_pool", [], "wipefs -p /dev/sda")
-        self._cmdEq("wipefs -p /dev/sdb", -2)
+        self._checkCmd("ssm remove default_pool", [], "wipefs -a -t btrfs /dev/sda")
+        self._cmdEq("wipefs -a -t btrfs /dev/sdb", -2)
 
-        self._checkCmd("ssm remove my_pool", [], "wipefs -p /dev/sdc3")
-        self._cmdEq("wipefs -p /dev/sdc1", -2)
-        self._cmdEq("wipefs -p /dev/sdc2", -3)
+        self._checkCmd("ssm remove my_pool", [], "wipefs -a -t btrfs /dev/sdc3")
+        self._cmdEq("wipefs -a -t btrfs /dev/sdc1", -2)
+        self._cmdEq("wipefs -a -t btrfs /dev/sdc2", -3)
 
         # remove subvolume
         self._addVol('vol001', 117283225, 1, 'default_pool', ['/dev/sda'], '/mnt/test')
@@ -189,24 +189,24 @@ class BtrfsFunctionCheck(MockSystemDataSource):
         self._addPool('other_pool', ['/dev/sdd', '/dev/sde'])
         self._checkCmd("ssm remove /dev/sdd /dev/sdb other_pool my_pool default_pool:/dev/default_pool/vol001", [],
             "btrfs subvolume delete /mnt/test")
-        self._cmdEq("wipefs -p /dev/sdc1", -2)
-        self._cmdEq("wipefs -p /dev/sdc3", -3)
-        self._cmdEq("wipefs -p /dev/sdc2", -4)
-        self._cmdEq("wipefs -p /dev/sde", -5)
-        self._cmdEq("wipefs -p /dev/sdd", -6)
+        self._cmdEq("wipefs -a -t btrfs /dev/sdc1", -2)
+        self._cmdEq("wipefs -a -t btrfs /dev/sdc3", -3)
+        self._cmdEq("wipefs -a -t btrfs /dev/sdc2", -4)
+        self._cmdEq("wipefs -a -t btrfs /dev/sde", -5)
+        self._cmdEq("wipefs -a -t btrfs /dev/sdd", -6)
         self._cmdEq("btrfs device delete /dev/sdb /mnt/test", -7)
         self._cmdEq("btrfs device delete /dev/sdd /tmp/mount", -8)
 
         self._removeMount("/dev/sda")
         # remove all
         self._checkCmd("ssm remove --all", [],
-            "wipefs -p /dev/sde")
-        self._cmdEq("wipefs -p /dev/sdd", -2)
-        self._cmdEq("wipefs -p /dev/sdc1", -3)
-        self._cmdEq("wipefs -p /dev/sdc3", -4)
-        self._cmdEq("wipefs -p /dev/sdc2", -5)
-        self._cmdEq("wipefs -p /dev/sda", -6)
-        self._cmdEq("wipefs -p /dev/sdb", -7)
+            "wipefs -a -t btrfs /dev/sde")
+        self._cmdEq("wipefs -a -t btrfs /dev/sdd", -2)
+        self._cmdEq("wipefs -a -t btrfs /dev/sdc1", -3)
+        self._cmdEq("wipefs -a -t btrfs /dev/sdc3", -4)
+        self._cmdEq("wipefs -a -t btrfs /dev/sdc2", -5)
+        self._cmdEq("wipefs -a -t btrfs /dev/sda", -6)
+        self._cmdEq("wipefs -a -t btrfs /dev/sdb", -7)
 
         # TODO
         # remove force
-- 
1.8.3.1

